lib/jobs/pdu-node-relations.js

```diff
@@ -0,0 +1,142 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+
+module.exports = pduRelationsJobFactory;
+di.annotate(pduRelationsJobFactory, new di.Provide('Job.Catalog.PduRelations'));
+di.annotate(pduRelationsJobFactory, new di.Inject(
+    'Job.Base',
+    'Services.Waterline',
+    'Logger',
+    'Util',
+    'Promise',
+    'Assert'
+));
+
+function pduRelationsJobFactory(
+    BaseJob,
+    waterline,
+    Logger,
+    util,
+    Promise,
+    assert
+) {
+
+    var logger = Logger.initialize(pduRelationsJobFactory);
+
+    /**
+     *
+     * @param {Object} [options]
+     * @constructor
+     */
+    function PduRelationsJob(options, context, taskId) {
+        PduRelationsJob.super_.call(this, logger, options, context, taskId);
+
+        this.nodeId = context.target || options.nodeId;
+        assert.isMongoId(this.nodeId);
+    }
+
+    util.inherits(PduRelationsJob, BaseJob);
+
+    /**
+     * Globals
+     */
+
+    var RELATION_TYPE = 'poweredBy';
+
+    /**
+     * @memberOf PduRelationsJob
+     */
+    PduRelationsJob.prototype._run = function run() {
+        var self = this;
+        return waterline.nodes.findByIdentifier(self.nodeId)
+        .then(function (node) {
+            if (node.type !== 'pdu') {
+                throw new Error('Task runs only on PDU nodes');
+            }
+            var pduHost = _adjustIpStrFormat(node.snmpSettings.host);
+            return self._searchObmServices(pduHost);
+        })
+        .then(function () {
+            self._done();
+        })
+        .catch(function (err) {
+            self._done(err);
+        });
+
+    };
+
+    PduRelationsJob.prototype._searchObmServices =
+        function _searchObmServices(pduHost) {
+            var self = this;
+            return Promise.try(function () {
```

Comment:
Ahh, I didn't realize this problem. the promise that returned from waterline isn't powerful than bluebird promise.